apps/03TicTacToe.js

```diff
@@ -22,23 +22,39 @@ function printBoard() {
 }
 
 function horizontalWin() {
-    // Your code here
+  if ((board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn) ||
+   (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) ||
+   (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn)) {
+    return true;
+  }
 }
 
 function verticalWin() {
-    // Your code here
+  if ((board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn) ||
+   (board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn) ||
+   (board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn)) {
+     return true;
+   }
 }
 
 function diagonalWin() {
-    // Your code here
+  if ((board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) ||
+   (board[2][0] === playerTurn && board[1][1] === playerTurn && board[0][2] === playerTurn)) {
+     return true;
+   }
 }
 
 function checkForWin() {
-    // Your code here
+    if (horizontalWin() === true || verticalWin() === true || diagonalWin() === true) {
+      console.log('Player ' + playerTurn + ' Won!');
+      return true;
+    }
 }
 
 function ticTacToe(row, column) {
-    // Your code here
+    board[row][column] = playerTurn;
+    checkForWin();
+    playerTurn = (playerTurn === 'X') ? 'O' : 'X';
```

Comment:
Everything looks great! When you get a chance, revisit this project and clear the board when someone wins, or prevent someone from putting an O or X where there is already a value. That's a good exercise to see how the program is running the commands and at what order it runs them.

Good work!
![allen](http://media3.giphy.com/media/Ph05xuYgrX5te/giphy.gif)